spec/extracted_code_spec.rb

```diff
@@ -0,0 +1,121 @@
+describe "Files and directories that shouldn't exist" do
+  let(:all_paths) { Dir.chdir(Rails.root) { Dir.glob("**/*") } }
+  let(:basenames) { all_paths.collect { |f| File.basename(f) } }
+
+  it "fully qualified paths" do
+    expect(all_paths).not_to include(
```

Comment:
How long do these take to run? I've ran into issues with tests like this which appear to run in O(n**2) time. Did you consider using a set difference?

Reply:
It takes less than 1 second on my machine.

Reply:
:+1: for set difference

Or you can use include_any? include_none? from more_core_extensions (which do set difference under the covers)